Document refactor cleanup

cmyr opened the pull request:
This is a first pass at fixing a bit of the mess left over from #162. 

While this PR looks a bit daunting there are very few actual changes. It is _mostly_ just moving things around.

- any code that was doing interaction handling or communication with the backend that could be trivially moved into the EditViewController was;

- the remaining code in EditView was organized by role, and got a little bit more documentation, to hopefully make it less daunting for other people to dip into;

- and a very few places (such as the code around blinking the cursor) were refactored to be a bit more concise and a bit more readable. 

- anything redundant or unused was removed. I hope I wasn't too gung-ho on this front. 😏 

Open & Save are now working as expected, fixing a brief regression.

There's still no cmd-T for a new tab. That's on deck, though. I'll open an issue. Tabs _do_ work when the tab bar is visible, and you can drag them out of and into windows.

raphlinus commented on XiEditor/EditView.swift:
```diff
@@ -500,168 +491,16 @@ class EditView: NSView, NSTextInputClient {
-    }
-
-    // Background color for selected text. Only the first responder of the key window should have a non-gray selection.
-    func selcolor() -> NSColor {
```
This PR is not compiling for me, selcolor() is still being referenced from EditView.draw().

cmyr replied:
grrr must've missed a push. I'm off for the night, will update tomorrow. 😒

raphlinus commented on XiEditor/EditView.swift:
```diff
@@ -198,7 +209,7 @@ class EditView: NSView, NSTextInputClient {
             attrString.addAttribute(NSForegroundColorAttributeName, value: randcolor, range: NSMakeRange(0, s.utf16.count))
             */
-            styleMap?.applyStyles(text: s, string: attrString, styles: line.styles, selColor: selcolor())
+            styleMap?.applyStyles(text: s, string: attrString, styles: line.styles, selColor: textSelectionColor)
```
Since we're drawing the selection rectangles rather than adding them as bgcolor attributes to the string, there's no need to pass the selection color into applyStyles.

raphlinus commented:
No worries, I'll get the threading cleanup in before then (will very likely require manual merging).

cmyr commented:
Sounds good, I will welcome to opportunity to hide my goof-up in a messy merge commit 😎

cmyr commented:
Alright, fixed that little issue and merged in your changes from last night, which look good.

There's one tiny program-flow change in the merge commit, which is that I moved the sending of pending notifications into a `didSet` on Document.tabName.

raphlinus commented:
I played with it, and took a pass over the code, and it looks good. Thanks!
